### PR TITLE
feat(handlers): add editChannelOverwrite() & deleteChannelOverwrite()

### PR DESCRIPTION
**Pull request by itohatweb**
**Please describe the changes this PR makes and why it should be merged:**

**Status**

- [x] Code changes have been tested against the Discord API, or there are no code changes

**Semantic versioning classification:**

- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/api/handlers/guild.ts [hunk 1]
```diff
@@ -206,6 +207,32 @@ export function swapChannels(
...
+  );
+  if (!hasPerm) {
+    throw new Error(Errors.MISSING_MANAGE_ROLES);
+  }
+
+  const body = {
+    allow: calculateBits(options.allow),
+    deny: calculateBits(options.deny),
+    type: options.type,
+  };
```
Comment (ayntee) [inline src/api/handlers/guild.ts:228]:
```suggestion
```

File: src/api/handlers/guild.ts [hunk 2]
```diff
@@ -206,6 +207,32 @@ export function swapChannels(
...
+
+  return RequestManager.put(
+    endpoints.CHANNEL_OVERWRITE(channelID, options.id),
+    body,
```
Comment (ayntee) [inline src/api/handlers/guild.ts:232]:
```suggestion
  {
    allow: calculateBits(options.allow),
    deny: calculateBits(options.deny),
    type: options.type,
  },
```

File: src/api/handlers/guild.ts (hunk 1 above)
Comment (ayntee) [reply src/api/handlers/guild.ts:228]: image
You can commit the suggestion using the "Commit suggestion" button :)

File: src/api/handlers/guild.ts [hunk 3]
```diff
@@ -206,6 +207,31 @@ export function swapChannels(
...
+  }
+
+
+  return RequestManager.put(
+    endpoints.CHANNEL_OVERWRITE(channelID, options.id),
+  {
+    allow: calculateBits(options.allow),
+    deny: calculateBits(options.deny),
+    type: options.type,
+  },
```
Comment (ayntee) [inline src/api/handlers/guild.ts:231]:
```suggestion
    {
      allow: calculateBits(options.allow),
      deny: calculateBits(options.deny),
      type: options.type,
    },
```

**Review by ayntee — COMMENTED** [review]
The ID of the overwrite is not arbitrary, so it is invalid for the ID to be in the `options` object parameter.

**Comment by itohatweb** [comment]
Should I add a new interface for that?

File: src/api/handlers/guild.ts [hunk 4]
```diff
@@ -206,6 +207,30 @@ export function swapChannels(
...
+
+  return RequestManager.put(
+    endpoints.CHANNEL_OVERWRITE(channelID, options.id),
+    {
+      allow: calculateBits(options.allow),
+      deny: calculateBits(options.deny),
+      type: options.type,
+    },
+  );
+}
```
Comment (ayntee) [inline src/api/handlers/guild.ts:232]:
```suggestion
export async function editChannelOverwrite(
  guildID: string,
  channelID: string,
  overwriteID: string,
  options: Pick<Overwrite, "allow" | "deny" | "type">,
) {
  const hasPerm = await botHasPermission(
    guildID,
    ["MANAGE_ROLES"],
  );
  if (!hasPerm) {
    throw new Error(Errors.MISSING_MANAGE_ROLES);
  }

  return RequestManager.put(
    endpoints.CHANNEL_OVERWRITE(channelID, overwriteID),
    {
      allow: calculateBits(options.allow),
      deny: calculateBits(options.deny),
      type: options.type,
    },
  );
}
```

File: src/api/handlers/guild.ts [hunk 5]
```diff
@@ -206,6 +207,31 @@ export function swapChannels(
   );
 }
 
+/** Edit the channel permission overwrites for a user or role in this channel. Requires MANAGE_ROLES permission. */
+export async function editChannelOverwrite(
+  guildID: string,
+  channelID: string,
+  overwriteID: string,
+  options: Pick<Overwrite, "allow" | "deny" | "type">,
```
Comment (ayntee) [inline src/api/handlers/guild.ts:215]:
```suggestion
  options: Omit<Overwrite, "id">,
```

**Review by ayntee — APPROVED** [review]
LGTM

**Comment by itohatweb** [comment]
Forgot one function: `deleteChannelOverwrite`

File: src/api/handlers/guild.ts [hunk 6]
```diff
@@ -206,6 +207,50 @@ export function swapChannels(
...
+    endpoints.CHANNEL_OVERWRITE(channelID, overwriteID),
+    {
+      allow: calculateBits(options.allow),
+      deny: calculateBits(options.deny),
+      type: options.type,
+    },
+  );
+}
+
+/** Delete the channel permission overwrites for a user or role in this channel. Requires MANAGE_ROLES permission. */
```
Comment (ayntee) [inline src/api/handlers/guild.ts:235]:
```suggestion
/** Delete the channel permission overwrites for a user or role in this channel. Requires `MANAGE_ROLES` permission. */
```

File: src/api/handlers/guild.ts [hunk 7]
```diff
@@ -206,6 +207,50 @@ export function swapChannels(
   );
 }
 
+/** Edit the channel permission overwrites for a user or role in this channel. Requires MANAGE_ROLES permission. */
```
Comment (ayntee) [inline src/api/handlers/guild.ts:210]:
```suggestion
/** Edit the channel permission overwrites for a user or role in this channel. Requires `MANAGE_ROLES` permission. */
```

**Review by ayntee — CHANGES_REQUESTED** [review]

**Review by ayntee — APPROVED** [review]
LGTM

**Pull request merged**